File: versioned_docs/version-1.10.x/examples/js/src/buckets_browse.mjs
import { Client, QuotaType } from "reduct-js";
import assert from "node:assert";

// Create a new client with the server URL and an API token
const client = new Client("http://127.0.0.1:8383", { apiToken: "my-token" });

// Browse the list of buckets and print their information
for (const info of await client.getBucketList()) {
  console.log(`Bucket: ${info.name}`);
  console.log(`Size: ${info.size}`);
  console.log(`Entry Count: ${info.entryCount}`);
  console.log(`Oldest Record: ${info.oldestRecord}`);
  console.log(`Latest Record: ${info.latestRecord}`);
}

// Get information about a specific bucket
const bucket = await client.getBucket("example-bucket");
const settings = await bucket.getSettings();
console.log(`Settings: ${settings}`);
for (const info of await bucket.getEntryList()) {
  console.log(`Entry: ${info.name}`);
  console.log(`Size: ${info.size}`);
  console.log(`Oldest Record: ${info.oldestRecord}`);
  console.log(`Latest Record: ${info.latestRecord}`);
}

File: docs/examples/go/src/buckets_browse.go
package main

/**

async def browse_buckets():
    # Create a client with the base URL and API token
    async with Client("http://localhost:8383", api_token="my-token") as client:
        # Browse all buckets and print their information
        buckets: List[BucketInfo] = await client.list()
        for info in buckets:
            print(f"Bucket: {info.name}")
            print(f"Size: {info.size}")
            print(f"Entry Count: {info.entry_count}")
            print(f"Oldest Record: {info.oldest_record}")
            print(f"Latest Record: {info.latest_record}")

        # Get information about a specific bucket
        bucket: Bucket = await client.get_bucket("example-bucket")
        info = await bucket.get_full_info()
        print(f"Bucket settings: {info.settings}")

        for entry in info.entries:
            print(f"Entry: {entry.name}")
            print(f"Size: {entry.size}")
            print(f"Oldest Record: {entry.oldest_record}")
            print(f"Latest Record: {entry.latest_record}")
*/
import (
	"context"
	reduct "github.com/reductstore/reduct-go"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Browse all buckets and print their information
	buckets, err := client.GetBuckets(context.Background())
	if err != nil {
		panic(err)
	}

	for _, info := range buckets {
		println("Bucket:", info.Name)
		println("Size:", info.Size)
		println("Entry Count:", info.EntryCount)
		println("Oldest Record:", info.OldestRecord)
		println("Latest Record:", info.LatestRecord)
	}

	// Get information about a specific bucket
	// wait: https://github.com/reductstore/reduct-go/issues/24
}
